src/components/DocumentViewer.tsx

```diff
@@ -1,33 +1,46 @@
 import React, { useEffect, useRef } from 'react';
 import useInstance from '../context';
 import WebViewer from '@pdftron/webviewer';
-import type { WebViewerOptions } from '@pdftron/webviewer';
+import type { WebViewerOptions, WebViewerInstance } from '@pdftron/webviewer';
 import PropTypes from 'prop-types';
 
 export type TProps = {
   className?: string
+  isSimpleDisplay?: boolean
 } & WebViewerOptions
 
 export type TRef = React.RefObject<HTMLDivElement>
 
 const DocumentViewer = React.forwardRef<TRef, TProps>(
-  ({ className, ...rest }, ref) => {
+  ({ isSimpleDisplay, className, ...rest }, ref) => {
     const localRef = useRef<HTMLDivElement>(null);
     const { instance, setInstance } = useInstance();
 
+    function prepareSimpleUi (instance: WebViewerInstance) {
+      const { documentViewer, DisplayMode, DisplayModes } = instance.Core;
+      const displayMode = documentViewer.getDisplayModeManager();
+      displayMode.setDisplayMode(new DisplayMode(documentViewer, DisplayModes.Single));
+      instance.UI.setFitMode(instance.UI.FitMode.FitPage);
+    }
```

Comment:
This code can be the SimpleDisplay component. Doesn't make sense to have it here

Reply:
Also functions inside React components must be wrapped in a useCallback

Reply:
How about exporting it as a function outside of the component from SimpleDisplay? I think we can saves the usage of useCallback this way. Also changed the name of the function as it not exactly "prepares simple UI" but changes display to single and fit mode.

Reply:
I would say what you implemented is actually worse than before 😆 

We shouldn't be coupling any simple viewer logic with the standard viewer.

I would suggest accepting a `onReady` in `DocumentViewer`, then using that inside SimpleViewer

function prepareSimpleUI (instance: WebViewerInstance) {
      const { documentViewer, DisplayMode, DisplayModes } = instance.Core;
      const displayMode = documentViewer.getDisplayModeManager();
      displayMode.setDisplayMode(new DisplayMode(documentViewer, DisplayModes.Single));
      instance.UI.setFitMode(instance.UI.FitMode.FitPage);
    }

```
const SimpleViewer = () => {
  return <DocumentViewer onReady={prepareUI} />
}
```

Reply:
Edited.